### PR TITLE
Add fossil-scm

### PR DESCRIPTION
**Pull request by sweegiap**
_Motivation:_  Explain here what the reason for the pull request is.
   To make 'fossil' SPK available.

_Linked issues:_  Optionally, add links to existing issues or other PR's
   Not Applicable  

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by sweegiap** [comment]
Build rule all-supported completed successfully
And successful installation of package (armadaxp on DS214). 

What is the "Package upgrade" completed successfully checklist item?

**Comment by hgy59** [comment]
@sweegiap thanks for contribution.
some thoughts of mine:
- You should not use the term "synocli" in the name of this package, since synocli is used for collections of multiple cli tools of a certain topic (and are too small to build a separate package for each of them)
- IMHO fossil should be a package of it's own (call it fossil or fossil-scm)
- fossil is not only cli, a quick lookup told me that it has a built in web interface, so I propose to integrate this too.
- as fossil has a kind of persitent data you should ensure that user/business data is not lost on package upgrade. While development you can easily create a new version by incrementing the SPK_REV. Then you install this version over an already installed package and verify that the data is still availlable after upgrade. If the sofware provides a kind of data migration, this should be integrated and validated too.
- it is also common to provide a backup location to save the data before an update. Therefore you need to add a UI-Wizard.

File: spk/synocli-fossil/Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+SPK_NAME = synocli-fossil
+SPK_VERS = 1.0
+SPK_REV = 3
```
Comment (hgy59) [inline spk/synocli-fossil/Makefile]: use 1 as initial value

File: spk/synocli-fossil/Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+SPK_ICON = src/synocli-fossil.png
+
+include ../../mk/spksrc.common.mk
```
Comment (hgy59) [inline spk/synocli-fossil/Makefile]: remove include of spksrc.common.mk, needed only when common variables are used before include of spksrc.spk.mk (like xxx_ARCHES)

File: spk/synocli-fossil/PLIST [hunk 3]
```diff
@@ -0,0 +1 @@
+bin:bin/fossil
```
Comment (hgy59) [inline spk/synocli-fossil/PLIST]: this should be moved to cross PLIST file

File: cross/fossil/Makefile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+PKG_NAME = fossil-scm
```
Comment (hgy59) [inline cross/fossil/Makefile]: it is common to name the folder like the package where applicable (use cross/fossil-scm  **and spk/fossil-scm**)

File: cross/fossil/Makefile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+GNU_CONFIGURE = 1
+CONFIGURE_ARGS = --static --json 
+CONFIGURE_ARGS += --disable-fusefs 
+CONFIGURE_ARGS += --with-th1-docs --with-th1-hooks 
+CONFIGURE_ARGS += --with-openssl=auto
+
+include ../../mk/spksrc.cross-cc.mk
+
+.PHONY: fixMakefileIn
+fixMakefileIn:
```
Comment (hgy59) [inline cross/fossil/Makefile]: please replace the name fixMakefileIn by `fossil-scm_pre_configure`

File: cross/fossil/Makefile [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+PKG_DIST_NAME = $(PKG_NAME)-v$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = http://www.fossil-scm.org/index.html/tarball/version-$(PKG_VERS)
+PKG_DIR = $(PKG_NAME)-v$(PKG_VERS)
+
+BUILD_DEPENDS = cross/zlib
+BUILD_DEPENDS += cross/openssl
+
+HOMEPAGE = http://www.fossil-scm.org/
+COMMENT  = Fossil is a simple, high-reliability, distributed software configuration management system.
+LICENSE  = http://www.fossil-scm.org/home/doc/trunk/COPYRIGHT-BSD2.txt
```
Comment (hgy59) [inline cross/fossil/Makefile]: for well known licenses place the type here: `LICENSE = BSD2`
And use the same in the spk Makefile

**Comment by sweegiap** [comment]
Updated pull request with recommended changes above in new commit (ac568bb).

**Comment by sweegiap** [comment]
> * fossil is not only cli, a quick lookup told me that it has a built in web interface, so I propose to integrate this too.
> * as fossil has a kind of persitent data you should ensure that user/business data is not lost on package upgrade. While development you can easily create a new version by incrementing the SPK_REV. Then you install this version over an already installed package and verify that the data is still availlable after upgrade. If the sofware provides a kind of data migration, this should be integrated and validated too.
> * it is also common to provide a backup location to save the data before an update. Therefore you need to add a UI-Wizard.

In my use case, I plan to use it with DSM ssh ([fossil clone](https://fossil-scm.org/home/help?cmd=clone)):
      fossil clone ssh://<user>@<dsm>/<path_to_repo>?fossil=/usr/local/bin/fossil  <local_clone>

At this point, I'm not familiar with how DSM services/UI development work at all :)
ps - fossil-scm repo file is essentially an sqlite database; and would not need changes when fossil version is upgraded.

**Comment by sweegiap** [comment]
Is there a way to either:
 - install fossil to /usr/bin 
 - create link /usr/bin/fossil to /usr/local/bin/fossil 

As /usr/local/bin is not in the default PATH for the ssh, fossil clone would always need the '?fossil=/usr/local/bin/fossil'.

File: spk/fossil-scm/Makefile [hunk 7]
```diff
@@ -0,0 +1,22 @@
+SPK_NAME = fossil-scm
+SPK_VERS = 1.0
```
Comment (hgy59) [inline spk/fossil-scm/Makefile]: use SPK_VERS = 2.10 (same as PKG_VERS)

File: spk/fossil-scm/Makefile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+SPK_REV = 1
+SPK_ICON = src/fossil-scm.png
+
+DEPENDS = cross/fossil-scm
+
+MAINTAINER = SynoCommunity
+DISPLAY_NAME = Fossil SCM
+
+OPTIONAL_DESC =
```
Comment (hgy59) [inline spk/fossil-scm/Makefile]: remove this variable as it is not used here (I see, you cloned a synocli-* Makefile)

File: spk/fossil-scm/Makefile [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+DESCRIPTION = "Fossil is a simple, high-reliability, distributed software configuration management system."
+STARTABLE = no
+CHANGELOG = "Initial version."
+
+HOMEPAGE = https://github.com/SynoCommunity/spksrc/wiki/FAQ-FossilSCM
```
Comment (hgy59) [inline spk/fossil-scm/Makefile]: the synocommunity wiki is used as homepage for synocli-* packages to collect the information of the individual cli tools.
Usualy this is the homepage/documentation of the packaged software.
use: https://fossil-scm.org/home/doc/trunk/www/index.wiki

File: spk/fossil-scm/Makefile [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+LICENSE  = Each tool is licensed under it\'s respective license.
```
Comment (hgy59) [inline spk/fossil-scm/Makefile]: use the same as in cross/fossil-scm/Makefile
(another synocli-* copy)

**Comment by hgy59** [comment]
And please change the title of this PR to something like:
Add fossil-scm

**Comment by hgy59** [comment]
> 
> Is there a way to either:
>     * install fossil to /usr/bin
>     * create link /usr/bin/fossil to /usr/local/bin/fossil
> 
No, package installation is always linked to /var/packages/{package-name}/target (original installation is in /volume1@appstore/{package-name}, but this depends on the volume used for installation).

You already use SPK_COMMANDS variable to create the link /usr/local/bin/fossil.
Links in /usr/bin are not recommended, as they are lost on each DSM update.

**Comment by sweegiap** [comment]
Updated above recommendations with commit (777f231).
Used "https://www.fossil-scm.org/" for home page.

**Comment by sweegiap** [comment]
@hgy59 Anything pending from me for this PR?

**Comment by ymartin59** [comment]
Packages have been published

**Pull request merged**